### ISSUE TITLE
[BUG] NPE on case (in a switch statement) resolving to empty-sequence()

### ISSUE THREAD

**Issue by PieterLamers** [eXist-db/exist#3560]
> To be able to better understand you problem, please add as much information as possible to this ticket. Always test your bugs against the latest stable release of exist. We cannot provide support for older versions here on GitHub. If the version of eXist that is experiencing the issue is more than 1 major version behind the most recent release, please consider posting a question on our mailing list. 

**Describe the bug**
NPE on `case` (in a `switch` statement) resolving to `empty-sequence()`

**Expected behavior**
No NPE, no error, same behavior as with `case ()`; 

**To Reproduce**

    let $a := 'a'
    let $test := ('a','b','c')
    let $test-two := ('d','e')
    return switch( $a)
      case $test[. eq $a] return 'one'
      case $test-two[. eq $a] return 'two'
      default return ()

returns correct result 'one'; with 
`let $a := 'd'`
one will get an NPE presumably because the parser does not like 
`$test[. eq $a]` resolving to an empty sequence.

note that `case () return 'one'` does not throw an error of any kind (although it does not match with $a being empty either).

**Context (please always complete the following information):**
 - OS: windows/linux
 - eXist-db version: 5.3-snapshot, 4.7.1
 - Java Version [e.g. Java8u261]

**Comment by joewiz** [comment]
I can reproduce the error reported here, using a build from the current HEAD of the develop branch eXist 5.3.0-SNAPSHOT 9537590bf7a9a36fca9fd8768eb9f583d2b8c750 20201010061646. BaseX and Saxon do not raise the error.

Minimal code to trigger the error:

```xquery
xquery version "3.1";

switch(1)
    case 2[. eq 1] return "foo"
    default return "bar"
```

The error raised in eXide:

> err:FOER0000: Unidentified error. An error has been raised by the query

The error in exist.log:

```txt
2020-10-12 11:34:03,641 [qtp1172237449-50] ERROR (XQueryServlet.java [process]:558) - null 
java.lang.NullPointerException: null
	at org.exist.xquery.SwitchExpression.eval(SwitchExpression.java:107) ~[exist-core-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71) ~[exist-core-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at org.exist.xquery.PathExpr.eval(PathExpr.java:279) ~[exist-core-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71) ~[exist-core-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at org.exist.xquery.XQuery.execute(XQuery.java:261) ~[exist-core-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at org.exist.xquery.XQuery.execute(XQuery.java:185) ~[exist-core-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
	at org.exist.http.servlets.XQueryServlet.process(XQueryServlet.java:494) ~[exist-core-5.3.0-SNAPSHOT.jar:5.3.0-SNAPSHOT]
        ...
```

That leads to this line of code: https://github.com/eXist-db/exist/blob/34767fa9b5f221504d4c02a107b4e973fdd4bd0f/exist-core/src/main/java/org/exist/xquery/SwitchExpression.java#L107

**Comment by dizzzz** [comment]
fixed by #3567
